woocommerce/woocommerce-android
# Fix the variation generation bottom sheet disappearing

Fixes #8193. Visually, it's a small change but a more substantial refactoring was required to make this work.

The PR changes the variation generation dialog into bottom sheet and utilizes the Jetpack Navigation, which automatically handles the state-saving on rotation. This resulted in the removal of a listener, which was replaced by a new bottom sheet ViewModel because the bottom sheet was used in multiple places. The tap events are handled using `navigateBackWithNotice()` calls.

[device-2023-04-28-163309.webm](https://user-images.githubusercontent.com/1522856/235177429-9ff4dff0-b74d-49c2-be17-4e9e9ad46056.webm)

**To test:**
1. Go to products
2. Tap or create a variable product
3. Tap on Add variations
4. (add new attributes and options, if there are none)
5. Tap on the Add variation button
6. Notice a bottom sheet dialog appears
7. Rotate the screen
8. Notice the bottom sheet doesn't disappear
9. Tap on Generate all variations
10. Notice the generation dialog appears
11. Tap cancel
12. Tap on Add variation button again
13. Notice the bottom sheet appears
14. Tap on Add new variation
15. Notice a variation is generated
